Objects/longobject.c

```diff
@@ -3614,9 +3614,13 @@ long_richcompare(PyObject *self, PyObject *other, int op)
 static void
 long_dealloc(PyObject *self)
 {
+#ifdef Py_LIMITED_API
+#ifndef Py_GIL_DISABLED
```

Comment:
I'm not sure it's possible for us to know when we're using the limited API. `Py_LIMITED_API` is something that affects `Python.h` at the compile-time of an extension module, but this happens during the compilation of CPython, so it will never be true.